File: fsharp/FsFind/FsFindLib/FindConfig.fs
namespace FsFindLib
open System
open System.IO

type FindConfig() =
    let getXfindConfigDir () =
        let defaultXFindConfigDir = Path.Join(FileUtil.GetHomePath(), ".config", "xfind")
        let xFindConfigDir = Environment.GetEnvironmentVariable("XFIND_CONFIG_DIR")
        if String.IsNullOrEmpty(xFindConfigDir) then
            defaultXFindConfigDir
        else
            xFindConfigDir

    member val FileTypesPath : string = "FsFindLib.Resources.filetypes.json" with get, set
    member val FindOptionsPath : string = "FsFindLib.Resources.findoptions.json" with get, set
    member val DefaultFindSettingsPath : string = Path.Join(getXfindConfigDir(), "settings.json") with get, set

File: csharp/CsFind/CsFindLib/FindConfig.cs
using System;
using System.IO;

namespace CsFindLib;

public class FindConfig
{
    public string FileTypesPath { get; private set; }
    public string FindOptionsPath { get; private set; }
    public string DefaultFindSettingsPath { get; private set; }

    public FindConfig()
    {
        FileTypesPath = "CsFindLib.Resources.filetypes.json";
        FindOptionsPath = "CsFindLib.Resources.findoptions.json";

        var homePath = FileUtil.GetHomePath();
        var defaultXFindConfigDir = Path.Join(homePath, ".config", "xfind");
        var xFindConfigDir = Environment.GetEnvironmentVariable("XFIND_CONFIG_DIR");
        if (string.IsNullOrEmpty(xFindConfigDir))
        {
            xFindConfigDir = defaultXFindConfigDir;
        }

        DefaultFindSettingsPath = Path.Join(xFindConfigDir, "settings.json");
    }
}